src/refactorings/change-signature/change-signature.ts

```diff
@@ -114,26 +115,47 @@ function updateCode(
         const args = node.arguments.slice();
         if (args.length) {
           newPositions.forEach((order) => {
-            const arg = node.arguments[order.value.startAt];
-            args[order.value.endAt] = arg;
+            if (order.value.startAt === -1) {
+              // Convert to a valid code.
+              // Without that will trigger invalid "Missing semicolon (n, n)"
+              // That error occurs only for literal objects like: {id: 1, ...}
+              const fakedBlockCode = `const faked = ${order.value.val}`;
+              const parsed = parse(fakedBlockCode);
+              const node = parsed.program.body[0];
+
+              if (t.isVariableDeclaration(node)) {
+                const variableDeclarator = node.declarations[0];
+                if (variableDeclarator.init) {
+                  args[order.value.endAt] = variableDeclarator.init;
+                }
+              }
+            } else if (order.value.endAt === -1) {
```

Comment:
I guess we can use that instead?

```suggestion
            } else if (hasRemovedTheParameter(order)) {
```